client/src/pages/SinglePost.jsx

```diff
@@ -1,33 +1,70 @@
-import React from 'react'
-import Edit from "../img/temp2.jpg"
-import Delete from "../img/temp3.jpg"
-import Menu from "../components/Menu"
-import { Link } from 'react-router-dom'
+import React, { useEffect, useState } from "react";
+import Edit from "../img/temp2.jpg";
+import Delete from "../img/temp3.jpg";
+import Menu from "../components/Menu";
+import { Link, useParams } from "react-router-dom";
+import axios from "axios";
 
 const Single = () => {
+  const [loaded, setLoaded] = useState(false); // whether image data has been received from axios
+  const [imageURL, setImageURL] = useState(); // image URL generated from image data
+  const { id } = useParams(); // post id
+  // var filename = "CloudClub.png";  // Hardcoded image key
```

Comment:
let's remove this

Reply:
I removed the hardcoded image. The variables in lines 9-11 are needed for the logic to display the image.